src/ape/utils/abi.py

```diff
@@ -285,45 +285,47 @@ def decode_value(self, abi_type, value):
+    >>> parse_type({'type': 'tuple', 'components': [{'type': 'uint256'}]})
+    ('uint256',)
+    >>> parse_type({'type': 'tuple', 'components': [{'type': 'uint256'}, {'type':\
+    'uint256'}]})
+    ('uint256', 'uint256')
+    >>> parse_type({'type': 'tuple', 'components': [{'type': 'uint256'}, {'type':\
+    'tuple', 'components': [{'type': 'uint256'}]}]})
+    ('uint256', ('uint256',))
+    >>> parse_type({'type': 'tuple[]', 'components': [{'type': 'uint256'}]})
+    [('uint256',)]
+    >>> parse_type({'type': 'tuple[]', 'components': [{'type': 'uint256'}, {'type':\
+    'tuple', 'components': [{'type': 'uint256[]'}]}]}) 
+    [('uint256', (['uint256'],))]
+    >>> parse_type({'type': 'tuple', 'components': [{'type': 'tuple[]',\
+    'components': [{'type': 'int[]'}]}, {'type': 'int'}]})
+    ([(['int'],)], 'int')
+    >>> parse_type({'type': 'int[][]'})
+    [['int']]
 
-def parse_type(output_type: str) -> Union[str, Tuple, List]:
-    if not output_type.startswith("("):
-        return output_type
-
-    # Strip off first opening parens
-    output_type = output_type[1:]
-    found_types: List[Union[str, Tuple, List]] = []
-
-    while output_type:
-        if output_type.startswith(")"):
-            result = tuple(found_types)
-            if "[" in output_type:
-                return [result]
-
-            return result
-
-        elif output_type[0] == "(" and ")" in output_type:
-            # A tuple within the tuple
-            end_index = output_type.index(")") + 1
-            found_type = parse_type(output_type[:end_index])
-            output_type = output_type[end_index:]
-
-            if output_type.startswith("[") and "]" in output_type:
-                end_array_index = output_type.index("]") + 1
-                found_type = [found_type]
-                output_type = output_type[end_array_index:].lstrip(",")
-
-        else:
-            found_type = output_type.split(",")[0].rstrip(")")
-            end_index = len(found_type) + 1
-            output_type = output_type[end_index:]
-
-        if isinstance(found_type, str) and "[" in found_type and ")" in found_type:
-            parts = found_type.split(")")
-            found_type = parts[0]
-            output_type = f"){parts[1]}"
-
-        if found_type:
-            found_types.append(found_type)
+    Args:
+        type (Dict[str, Any]): This is ABIType.dict(). The two keys
+        that we are interested in are `type` and `components`.
+        `components` produces the same two keys (if present),
+        namely `type` and `components`.
 
-    return tuple(found_types)
+    Returns:
+        Union[str, Tuple, List]: Python representation of the type.
+        For example, `uint256[]` becomes ['uint256'].
+    """
+    if type["type"] == "tuple[]":
+        return [tuple([parse_type(c) for c in type["components"]])]
+    elif type["type"] == "tuple":
+        return tuple([parse_type(c) for c in type["components"]])
+    elif type["type"].endswith("[][]"):
```

Comment:
What about `[][][]` or `[][][][]`, does those still work?
Not that are common data structures in contracts, but programmatically the design should handle it